components/Layout/Header/Header.jsx

```diff
@@ -1,57 +1,96 @@
+import { useState, useEffect } from 'react';
+import cx from 'classnames';
 import Link from 'next/link';
+import { useRouter } from 'next/router';
+
 import { useContext } from 'react';
 import UserContext from 'components/UserContext/UserContext';
+import { getDataIncludes } from 'utils/saveData';
+import Logo from './Logo.jsx';
+
+const loggedNavLinks = [
+  {
+    name: 'Search',
+    path: '/',
+  },
+  {
+    name: 'My cases',
+    path: '/cases?my_notes_only=true',
+  },
+  {
+    name: 'Forms in progress',
+    path: '/form-in-progress',
+  },
+  {
+    name: 'Logout',
+    path: '/logout',
+  },
+];
 
 const HeaderComponent = ({ serviceName }) => {
   const { user } = useContext(UserContext);
-
-  const userDetails = (
-    <div className="govuk-custom-header__links">
-      <p>
-        <span className="govuk-!-font-weight-bold">Logged in as:</span>{' '}
-        {user && user.name}
-      </p>
-      <a className="govuk-!-font-weight-bold" href="/logout">
-        Logout
-      </a>
-    </div>
-  );
+  const { asPath } = useRouter();
+  const [navLinks, setNavLinks] = useState(loggedNavLinks);
+  useEffect(() => {
+    if (!user) {
+      setNavLinks();
+    } else if (!getDataIncludes('/form')) {
+      setNavLinks(navLinks.filter(({ name }) => name !== 'Forms in progress'));
```

Comment:
Getting an undefined error when navigating from the confirmation page to the home page

Reply:
Where exactly? I can't reproduce it 🤔

Reply:
Fix working